### ISSUE TITLE
How to build

### ISSUE THREAD

**Issue by gaurav3017** [kiwibrowser/android#12]
I am new to android development and bit unsure how to build this project as the project structure doesn't seem to be "generic" Android Studio project. Any documentation etc would be really helpful.

**Comment by Madis0** [comment]
As it says on the readme:

> We do not provide support or help on how to use or build the source-code.

You might want to try the [instructions for building Bromite](https://github.com/bromite/bromite#how-to-build) with Kiwi.

**Comment by csagan5** [comment]
My 2 cents: the bigger issue is that this repository here does not contain all the patches needed to build Kiwi :) so Kiwi is **not** open source, but by having the github repository here it looks like it is.

I think it would be fair to mention somewhere that the source code used to build Kiwi is not available anywhere (which is required to call it open source); I have seen it already mentioned together with Bromite as "open source browsers" in many places online, hence my comment here.

Cheers :hibiscus:

**Comment by dimqua** [comment]
@csagan5 How do you know that? They don't need to release patches to call it open source, the full source code tree is ok too.

**Comment by csagan5** [comment]
@dimqua sure, either full source code or patches (although without patches or commits for such a big project it's a lot of work to re-use the code, we are talking several hours time).

I concluded after some searching that the patches in https://github.com/kiwibrowser/privacy_patches are not here, but there is a more precise approach if you spend some time (something that bloggers do not do, apparently :laughing: colour me surprised).

The more precise approach is to check with a diff against the upstream Chromium tree. The version used in Kiwi supposedly is here (not stated anywhere and no tags corresponding to Chromim): https://github.com/kiwibrowser/android/blob/eb2015c28925be91b4a3130b3c2bee2f5edc91de/chrome/VERSION `69.0.3469.0` (referring to commit eb2015c28925be91b4a3130b3c2bee2f5edc91de, which is currently the only single commit)

Then make a diff with upstream Chromium `69.0.3469.0`; I have checked each tag available, the results are available here (including a sample script): https://gist.github.com/csagan5/0d080d4b770af6d82807d16ffb9cbf5a

This is the summary of the changes: https://gist.github.com/csagan5/0d080d4b770af6d82807d16ffb9cbf5a#file-summary-txt

```
# https://github.com/kiwibrowser/android/tree/4caf9f6585
 create mode 100644 ../kiwibrowser/chrome/browser/ui/cocoa/media_picker/create_desktop_media_picker_cocoa.h
 create mode 100644 ../kiwibrowser/chrome/browser/ui/cocoa/media_picker/create_desktop_media_picker_cocoa.mm
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/mouse-wheel-main-frame-scroll-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/latched-scroll-node-removed.html
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheel-in-scrollbar-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheelevent-basic-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheelevent-in-horizontal-scrollbar-in-rtl-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheelevent-in-text-node-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheelevent-in-vertical-scrollbar-in-rtl-expected.txt

# https://github.com/kiwibrowser/android/tree/eb2015c28925be91b4a3130b3c2bee2f5edc91de
 create mode 100644 ../kiwibrowser/chrome/browser/ui/cocoa/media_picker/create_desktop_media_picker_cocoa.h
 create mode 100644 ../kiwibrowser/chrome/browser/ui/cocoa/media_picker/create_desktop_media_picker_cocoa.mm
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/mouse-wheel-main-frame-scroll-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/latched-scroll-node-removed.html
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheel-in-scrollbar-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheelevent-basic-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheelevent-in-horizontal-scrollbar-in-rtl-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheelevent-in-text-node-expected.txt
 create mode 100644 ../kiwibrowser/third_party/WebKit/LayoutTests/fast/events/wheel/wheelevent-in-vertical-scrollbar-in-rtl-expected.txt
```

So basically it's even worse than I originally thought, there is absolutely nothing in this Github repository (a test for the layout, but that does not count).

**Comment by dimqua** [comment]
Wow, thanks for the so the detailed and informative answer! 

> if you would manage to build it, you would get simply Chromium

Unless you would apply the very few patches from https://github.com/kiwibrowser/privacy_patches, but it's clearly not enough for calling the whole browser "open source".

**Comment by projectextremum** [comment]
If it contains some patches that ain't open source, it is enough to call it **not open source**.

@csagan5 thanks for the analysis. Although it seems harsh, it appears right.

**Comment by csagan5** [comment]
@projectextremum it is reproducible, you can verify by yourself. My whole point is not against author but against people that do not spend any time to verify anything. In this particular case the app seems to not have anything malicious in the APK but in other countless occasions you will install malware because it has a Github repo, think about it. I am against this type of trust.

**Comment by Gagnef03** [comment]
I don't think you can, Arnaud has said that the whole source code isn't available (since he can't open source everything)

**Comment by csagan5** [comment]
> he can't open source everything

Why is that? Do you have a source for this information?
